2.x/SCO/x-oca-asset.md

```diff
@@ -0,0 +1,289 @@
+It can be referenced to give context on events that took place on this asset.
+It describes the hostname, the associated IP addresses and physical addresses of the asset.
+In SIEM terminology which collects events from many systems - it is used to describe where the events took place.
+It is referenced in the [`x-oca-event`](./x-oca-event.md) object.
+
+While `x-oca-asset` specifies the host on which the associated event occurred, it may provide extenstions to describe addtional entities of tne runtime environment. In a cloud environment the event may occur within a container that is running or a partcular pod on the host. `x-oca-asset` uses a separate extension for each addtional runtime entity. Please note that as per the STIX [2.0](http://docs.oasis-open.org/cti/stix/v2.0/cs01/part3-cyber-observable-core/stix-v2.0-cs01-part3-cyber-observable-core.html#_Toc496715388)and [2.1](https://docs.oasis-open.org/cti/stix/v2.1/csprd01/stix-v2.1-csprd01.html#_Toc16070830), COOs/SCOs can have any number of extenstions.
+
+| property name | type | description |
+|--|--|--|
+| **type** (required) | `string` | x-oca-asset |
+| **id** (required) | `string` | A valid [stix-id](https://docs.oasis-open.org/cti/stix/v2.1/os/stix-v2.1-os.html#_64yvzeku5a5c) based on`x-ibm-ttp-tagging` |
+| **spec_version** (optional) | `string` | The value of this property MUST be 2.1 for STIX Objects defined according to this specification. |
+| **extensions** | `dictionary` | Specifies any extensions of the object, as a dictionary. |
+| **device_id** | `string` | The ID of the device. |
+| **hostname** (required)|`string`|name of this host|
+| **ip_refs**|`list` of type `object-ref`| references the ip addresses related to this host. must be `ipv4-addr` or `ipv6-addr`|
+| **mac_refs**|`list` of type `object-ref`| references the mac addresses related to this host. must be of type `mac-addr`|
+| **os_ref**| `object-ref` | Specifies the operating system of the asset, as a reference to the Cyber-observable Object. The object referenced must be of type `software`. |
+| **architecture** | `string` | Architecture of the asset. |
+| **uptime** | `string` | Specifies the seconds the asset has been up.|
+| **host_type** | `string` | Specifies the type of asset. Example, firewall, t2.micro etc.|
+| **ingress** | `x-oca-traffic-type` | Specifies information like interface number and name, vlan, and zone information to classify ingress traffic using the Traff9c-type sub-object.|
+| **egress** | `x-oca-traffic-type` | Specifies information like interface number and name, vlan, and zone information to classify egress traffic using the Traffic-type sub object.|
+| **geo_ref**|`object-ref`| references the geolocation of this host. must be of type `x-oca-geo`|
+
+### Container Extenstion
+
+Extension Name `x-oca-container-ext`
+
+The container asset extension represents a container.
+
+| property name | type | description |
```

Comment:
Probably falls out of scope for what you are trying to do, but these descriptions aren't great. They are obvious (except for container type which is somewhat vague) so not a big deal or anything.

Reply:
Wow, great pick. 

This field is a clear example of the need for controlled language, and hence should actually be either an Enum (fixed set of strings, cannot be added to) or a Vocab (set of strings, can be added to).

Can you give us a list of strings/descriptions for this type field @subbyte , and your choice of Enum versus Vocab??

It will be easy to add a small table to the markdown doc, and create support in our [OCA Vocab repo](https://github.com/os-threat/stix2-dialect-definitions/tree/main/oca/vocab)